### ISSUE TITLE
Invalid  xcactivitylog file

### ISSUE THREAD

**Issue by chatwyn** [MobileNativeFoundation/XCLogParser#139]
Hi, When I use Xcode to build, It will generate xcactivitylog.
But  When I use xcodebuild cli, For example

```
xcodebuild -workspace xx.xcworkspace -scheme xx -sdk iphoneos -showBuildTimingSummary -resultBundlePath ./a/a -derivedDataPath ./a clean build
```

it will generate two xcactivitylog, One is a clean job log, and one is a build job log.  But the build job log is not a valid gzip. 
I have to wait one day, but nothing changes.  
 
So, Is it possible to crash in generate xcactivitylog？

**Comment by patrickbdev** [comment]
Similar to myself however for me it depends on the size of the project.
If I create a new xcode project, `xcodebuild` works.
If I use it on an existing large xcode project, I receive `.xcactivitylog is not a valid xcactivitylog file`.
Even after waiting an hour and running `xclogparser`, I get the same error.
This is my command:
```
xcodebuild \
-workspace XXX.xcworkspace \
-scheme XXX \
-destination generic/platform=iOS \
-resultBundlePath ~/Desktop/BuildTimeReports/BuildLogs/Result.xcresult \
clean build-for-testing \
OTHER_SWIFT_FLAGS="-Xfrontend -debug-time-function-bodies -Xfrontend -debug-time-expression-type-checking"
```

**Comment by chatwyn** [comment]
Yes, I found if I use hmap replace header search path,  it can generate .xcactivitylog correctly.    So I think it may be an `argument to long` error in xcodebuild.     But I don't know how to create issue to xcodebuild.

**Comment by patrickbdev** [comment]
> Yes, I found if I use hmap replace header search path,  it can generate .xcactivitylog correctly.    So I think it may be an `argument to long` error in xcodebuild.     But I don't know how to create issue to xcodebuild.

Thanks for responding, could you give a bit more information on what "hmap replace header search path" is?

**Comment by patrickbdev** [comment]
I've switched to using Xcode directly to build instead of `xcodebuild` and that works. I do need to wait a few seconds after the build is finished before running xclogparser.

**Comment by chatwyn** [comment]
https://tech.meituan.com/2021/02/25/cocoapods-hmap-prebuilt.html

**Comment by chatwyn** [comment]
me too. I think is xcodebuild bug,  but I don't know how to  submit bug to xcodebuid

**Comment by stevelandeyasana** [comment]
I had this problem as well, and I "fixed" it by adding `sleep 2` to the script that runs xclogparser.

**Comment by imWildCat** [comment]
I'm seeing the same problem as well using Xcode Version 13.2.1 (13C100).

`xcodebuild` with incremental builds is fine. But clean builds didn't work at all.

**Comment by vmzhivetyev** [comment]
any updates?

**Comment by Sherlouk** [comment]
Just adding as a user of XCMetrics (which relies on this project), I too am running into problems where the underlying `xcactivitylog` is (for some reason) invalid on a medium sized project. Works fine via Xcode, but problematic via xcodebuild.

Open to any workarounds or fixes to this. Xcode 14.1.

**Comment by BalestraPatrick** [comment]
As written above, this seems to be because Xcode closes the file handler or similar way too early without letting the file writer flush the contents of the file completely to disk. This should be a bug report to Xcode, and you can workaround it as described above by adding a sleep or running another action (such as a `clean` action) after your build.

**Comment by Sherlouk** [comment]
> such as a `clean` action

Feels extremely janky, but this has worked for us. Though this exasperate an issue on XCMetrics where clean builds are pushing up their logs.
